src/spikeinterface/sortingcomponents/benchmark/benchmark_matching.py

```diff
@@ -16,101 +17,483 @@
+        we_kwargs: dict
+            A dictionary of keyword arguments for the WaveformExtractor.
+        template_mode: {'mean' | 'median' | 'std'}
+            The mode to use to extract templates from the WaveformExtractor. (Default: 'median')
+        **kwargs
+            Keyword arguments for the run_matching method.
+
+        Returns
+        -------
+        matching_df : pandas.DataFrame
+            A dataframe of NumpySortings for each method/parameter_value/iteration combination.
+        """
+        try:
+            run_matching_fn = self.parameter_name2matching_fn[parameter_name]
+        except KeyError:
+            raise ValueError(f"Parameter name must be one of {list(self.parameter_name2matching_fn.keys())}")
+        try:
+            progress_bar = self.job_kwargs["progress_bar"]
+        except KeyError:
+            progress_bar = False
+        try:
+            assert isinstance(num_replicates, int)
+            assert num_replicates > 0
+        except AssertionError:
+            raise ValueError("num_replicates must be a positive integer")
+
+        sortings, gt_sortings, parameter_values, parameter_names, iter_nums, methods = [], [], [], [], [], []
+        if progress_bar:
+            parameters = tqdm(parameters, desc=f"Vary Parameter ({parameter_name})")
+        for parameter in parameters:
+            if progress_bar and num_replicates > 1:
+                replicates = tqdm(range(1, num_replicates + 1), desc=f"Replicating for Variability")
+            else:
+                replicates = range(1, num_replicates + 1)
+            for i in replicates:
+                sorting_per_method, gt_sorting = run_matching_fn(
+                    parameter, seed=i, we_kwargs=we_kwargs, template_mode=template_mode, **kwargs
+                )
+                for method in self.methods:
+                    sortings.append(sorting_per_method[method])
+                    gt_sortings.append(gt_sorting)
+                    parameter_values.append(parameter)
+                    parameter_names.append(parameter_name)
+                    iter_nums.append(i)
+                    methods.append(method)
+                if running_in_notebook():
+                    from IPython.display import clear_output
+
+                    clear_output(wait=True)
+                    for bar in progress_bars:
+                        display(bar.container)
+                    display(parameters.container)
+                    if num_replicates > 1:
+                        display(replicates.container)
+        matching_df = pd.DataFrame(
+            {
+                "sorting": sortings,
+                "gt_sorting": gt_sortings,
+                "parameter_value": parameter_values,
+                "parameter_name": parameter_names,
+                "iter_num": iter_nums,
+                "method": methods,
+            }
+        )
+        return matching_df
```

Comment:
Why a DataFrame here ?
Is it for further selection  on one paritcular columns ?